### DISCUSSION TITLE
Question: any hints on how to configure so i could use authelia or keycloak?

### DISCUSSION THREAD

**Discussion by airdogvan** [nginx-proxy/nginx-proxy#2223] [Q&A] [answered]
If too complicated then let's drop it.

Thanks

**Comment by buchdag** [comment]
Hi. Could you provide a bit more info beside _"how to configure so i could use authelia or keycloak"_, which does not tell us much about what you want to do ?

**Comment by airdogvan** [comment]
Sure.  Acme-companion for me is basically the gateway to about 20 services running in containers.
Each service (nextcloud, wallabag, kanboard, etc) asks for userid and password and it's cumbersome to every time enter those.

My understanding is that both authelia or keycloak can help by having me enter those once and would identify me to all other services.
Maybe my understanding is wrong and I apologize if I didn't understand their docs correctly.

**Comment by buchdag** [comment, marked as answer]
Unfortunately that question isn't really relevant to this project, you'll have to consult the documentation of each of your individual services to know how to configure each one to use an identity provider like Keycloak.

That's not done at the reverse proxy level.

**Comment by airdogvan** [comment]
Seems to me it is relevant as the proxy has to tell every incoming connection to first go to the identifying portal (authelia or keycloak) before redirecting to other services.

Please refer to this website: https://itnext.io/nginx-as-reverse-proxy-in-front-of-keycloak-21e4b3f8ec53

He modifies ngninx to add     

location /auth {
      proxy_pass http://mykeycloak:8080;

Further comments from that site immediately related to this specific issue:
"Note, that if the client requests the endpoint with the path /auth Nginx passes the request to the internal network endpoint mykeycloak:8080/auth. This internal network endpoint should match the name of the service we defined earlier in the docker-compose file. Port 8080 is the default port exposed by the JBoss Keycloak image.

Last, but not least by setting proxy_set_header we pass to Keycloak information about the original host and client IP. This enables Keycloak to generate correct in-app navigation links. If we haven’t done this when a user clicks on the login button, the Keycloak page would send him to incorrect [https://mykeycloak:8080/auth/admin/master/console](http://localhost/auth/admin/master/console), instead of the correct [https://example.com/auth/admin/master/console](http://localhost/auth/admin/master/console) . We also want to pass the client IP to Keycloak to support proper access logging."

Your nginx config is dynamic as it has to redirect new containers as they come online.

As you can see further up his post, he mounts nginx.conf as a volume which allows him to modify it according to his needs.  But in the case of acme-companion I can't do this, can I?  But seems to me there should be a  a way to add it somewhere.

I'm not familiar enough with nginx to begin modifying your config that has been working wonderfully for me for the last 3 years (steady as a rock).

I thought you would be the best people to give me hints as to how to do this, if doable at all.

Thanks

**Reply by buchdag** [reply]
I'll transfer the discussion to nginx-proxy/nginx-proxy which handle the actual reverse proxying. The companion is only for obtaining SSL certificates through ACME compatible CA.

**Comment by airdogvan** [comment]
By reading further keycloak docs I'm realizing that this has nothing to do with the proxy.  Seems every single website has to be individually configured to interact with keycloack.
Not what I thought.  
Closing the discurssion.
